corehq/apps/user_importer/importer.py

```diff
@@ -389,11 +391,37 @@ def create_or_update_users_and_groups(domain, user_specs, group_memoizer=None, u
                         user.reset_locations(location_ids, commit=False)
 
                 if role:
-                    user.set_role(domain, roles_by_name[role].get_qualified_id())
+                    role = roles_by_name[role].get_qualified_id()
+                    user.set_role(domain, role)
+
+                if web_user:
+                    user.user_data.update({'login_as_user': web_user})
 
                 user.save()
 
-                if send_account_confirmation_email:
+                if web_user:
+                    current_user = CouchUser.get_by_username(web_user)
+                    if not current_user and is_account_confirmed:
+                        raise UserUploadError(_(
+                            f"You can only set 'Is Account Confirmed' to 'True' on an existing Web User."
+                        ))
+                    if current_user and not current_user.is_member_of(domain) and is_account_confirmed:
+                        current_user.add_as_web_user(domain, role=role, location_id=user.location_id)
+                    elif not current_user or not current_user.is_member_of(domain):
+                        invite_data = {
+                            'email': web_user,
+                            'invited_by': 'Mobile User Upload',
+                            'invited_on': datetime.utcnow(),
+                            'domain': domain,
+                            'role': role,
+                            'supply_point': user.location_id
+                        }
+                        invite = Invitation(**invite_data)
+                        invite.save()
+                        if send_account_confirmation_email:
+                            invite.send_activation_email()
+
+                if send_account_confirmation_email and not web_user:
```

Comment:
I think I sort of answered this on the following comment, but yeah, this can now be used for either web user or mobile user confirmation, but you wouldn't do both. I think we will likely no longer need the email for the mobile user, but it is still configurable in the file. Name and password do not change, but it is believable they will no longer need to be shared with the end user, who will only interact with the webuser

Reply:
Are you following up on removing the mobile user confirmation if we don't need it anymore? Obviously not high priority, just don't want it to get forgotten about.

Reply:
How do mobile worker passwords work in this scenario? Just wanted to ensure we aren't leaving any potential security holes even though we don't expect direct login. I guess they are managed like all other accounts in the domain so the passwords are provided on upload, checked by the domain's rules, etc.? Just asking because `is_account_confirmed=False` allows you to bypass password rule checking I think

Reply:
> Are you following up on removing the mobile user confirmation if we don't need it anymore?
I had left it in as an option in case future projects want to use it, and will not be using web users for parts of their. It is conceivable to me that the mobile worker workflow would be useful to some future projects that have a single project space and want to do all of this by mobile worker not web user. We might even still want this for the single county users in New York. I am not totally sure whether they are switching to web users as well.

> How do mobile worker passwords work in this scenario
The presence of a web user in the row bypasses all of the work that you did, so those rules would be enforced for mobile users in the upload even with that column if there is a web user. Otherwise, it would have the behavior you added.